core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/batchmanager/DefaultBatchManager.java

```diff
@@ -129,13 +133,17 @@ private void handleAndCompleteResponses(BatchResponseT batchResult, Throwable ex
             requests.forEach((contextId, batchExecutionContext) -> batchExecutionContext.response()
                                                                                         .completeExceptionally(exception));
         } else {
-            List<IdentifiableMessage<ResponseT>> identifiedResponses = responseMapper.mapBatchResponse(batchResult);
-            for (IdentifiableMessage<ResponseT> identifiedResponse : identifiedResponses) {
-                String id = identifiedResponse.id();
-                ResponseT response = identifiedResponse.message();
-                requests.get(id)
-                        .response()
-                        .complete(response);
+            List<Either<IdentifiableMessage<ResponseT>, IdentifiableMessage<Throwable>>> identifiedResponses =
+                responseMapper.mapBatchResponse(batchResult);
+            for (Either<IdentifiableMessage<ResponseT>, IdentifiableMessage<Throwable>> response : identifiedResponses) {
```

Comment:
Replaced the forEach loop with `responseMapper.mapBatchResponse(batchResult).forEach(...)`. Didn't use map since we didn't need to modify the underlying elements.